src/redprl/judgment.fun

```diff
@@ -27,6 +27,7 @@ struct
            in
              (([],[]), tau)
            end
+       | MATCH_RECORD (lbl, _) => (([],[]), RedPrlSortData.EXP)
```

Comment:
@jonsterling One trick, which doesn't always work, is to define a structure with just the datatype, use that to constrain the signature, and then build up the real structure constrained by the signature:
```
structure FooDT =
struct
  datatype t = A | B | C
end
signature FOO =
sig
  datatype t = datatype FooDT.t
  val toString: t -> string
end
structure Foo : FOO =
struct
  datatype t = datatype FooDT.t
  fun toString x =
    case x of A => "A" | B => "B" | C => "C"
end
```
Of course, if your datatype really depends upon other components that are hard to define outside the module, this doesn't always work.

You can attempt similar tricks that define the whole structure first, then define the signature with the appropriate datatype replications, and then rebind the structure constrained by the signature (shadowing the original):
```
structure Foo = struct ... end
signature FOO = sig datatype t = datatype Foo.t ... end
structure Foo : FOO = Foo
```
This works in MLton (with MLB), but not in SML/NJ (with CM, IIRC, because CM gets a little confused by two definitions of `Foo`.)  Technically SML doesn't allow for:
```
local
structure Foo = struct ... end
in
signature FOO = sig datatype t = datatype Foo.t ... end
structure Foo : FOO = Foo
end
```
because signatures are only allowed at the very top-level.  SML/NJ allows the above and one can achieve much of the same effect with MLBs, but, unfortunately, not with the exact same source code for the two systems.